### PR TITLE
Snapshot testing for roboto condensed

### LINKED ISSUE #30: Add Snapshot Testing
**Description**

The goal is to add test visual tests for all the components to ensure that the designs are not changed with change in code.
**Problems solved**

Tests make it easy to identify breaking changes and fix them as and when they occur.

**Alternatives**

**Additional context**

**Want to take up?**

Yes

### LINKED ISSUE #31: Add bold (700) weight for Roboto Condensed
**Description**
Vocabulary uses Roboto Condensed weight 700. It's not added at present but it should be. The goal is to add the weight and all associated documentation.

**Additional context**
Weight 400 was added in #5.

### PR DESCRIPTION
**Pull request by kushaggarwal**
## Fixes

Fixes #30  by @dhruvkb 

## Description

This PR aims at the integration of the snapshot test for robot condensed component in the font library

## Technical details

The implementation for the same is based upon visual testing for the components using the jest-puppeteer following the testing guidelines for jest

## Tests

To run the test the following command is to be run `npm run test:snapshot`
## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — CHANGES_REQUESTED** [review]
The infrastructure for snapshot testing has been added, so remove the supporting files, leaving only the specific tests and their associated snapshots.

**Comment by kushaggarwal** [comment]
Okay @dhruvkb thank you very much

File: package.json [hunk 1]
```diff
@@ -58,5 +58,12 @@
...
   ],
   "license": "MIT",
-  "repository": "github:creativecommons/fonts"
+  "repository": "github:creativecommons/fonts",
+  "dependencies": {
+    "jest": "^25.2.7",
+    "jest-image-snapshot": "^3.0.1",
+    "jest-puppeteer": "^4.4.0",
+    "puppeteer": "^2.1.1",
+    "start-server-and-test": "^1.10.11"
```
Comment (dhruvkb) [inline package.json]: These already exist as `devDependencies`. Remove from here.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Before moving on to the actual review, here's an issue you need to fix.

File: package.json (hunk 1 above)
Comment (kushaggarwal) [reply package.json]: Okay @dhruvkb will update the following

**Comment by kushaggarwal** [comment]
@dhruvkb 
I have successfully written the tests for roboto condensed but unable to run it, being new with Docker I am not able to understand as if how to proceed so, it would be really helpful if you could guide me with some pointers for the same.
Thanks

**Comment by dhruvkb** [comment]
@kushaggarwal start with [Docker's quickstart guide](https://docs.docker.com/get-started/).

**Comment by kushaggarwal** [comment]
Okay @dhruvkb will go through the same 
Thank You

**Comment by kushaggarwal** [comment]
Hello @dhruvkb 
While building the project using docker as per the docker start guide I am having this error 
Can you please help
Screenshot 2020-04-16 at 6 46 49 AM

**Comment by dhruvkb** [comment]
@kushaggarwal run the commands from the root directory.

**Comment by kushaggarwal** [comment]
Hello @dhruvkb 
My apologies for delaying the completion of this pull request 
When running from the root directory it still gives the same error the command I am using for the same is `docker build docker/puppeteer --tag font:1.0`.
Please help

**Comment by kushaggarwal** [comment]
Hello @dhruvkb 
My apologies for delaying the completion of this pull request
Please find the completed snapshot test for Roboto Condensed font
Thanks

**Comment by kushaggarwal** [comment]
Also @dhruvkb I suppose the test for the Source Sans pro havent been added till now
Please advice if I can proceed towards the addition of that too
Thanks

**Comment by kushaggarwal** [comment]
Hello @dhruvkb 
Please review the above pull request for addition of snapshot test for source sans font

**Review by dhruvkb — APPROVED** [review]

**Comment by Dishebh** [comment]
> Hello @dhruvkb
> While building the project using docker as per the docker start guide I am having this error
> Can you please help
> Screenshot 2020-04-16 at 6 46 49 AM

@kushaggarwal @dhruvkb I am facing the same issue. I am trying to run the command 
`sudo docker build --tag test1:0 .` from the root directory, and I get the following error:

`unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/dishebh/Desktop/fonts/Dockerfile: no such file or directory`

I get that it is not able to find any Dockerfile file / directory in the root folder, but even when I run it in the `fonts/docker/fonts/` directory (where a Dockerfile is present), I get the following error:

`COPY failed: stat /var/lib/docker/tmp/docker-builder438192968/package.json: no such file or directory`

saying it is unable to find `package.json` file. 

Can you please help me with it? Thanks!

**Comment by kushaggarwal** [comment]
@Dishebh 
Run the following commands from the root folder itself
`docker build --tag puppeteer:latest --file docker/puppeteer/Dockerfile .`
`touch docker/puppeteer/env_vars`
`docker/puppeteer/run.sh -it npm run snapshot`
As soon as you enter the docker container you can run the command
`npm run test:snapshot`
To run the snapshot test

**Comment by Dishebh** [comment]
@kushaggarwal thanks a lot for this! The container is now running. But, when I run the tests, `npm run test:snapshot`, I get the error:
`net::ERR_CONNECTION_REFUSED at http://localhost:8080/iframe.html?id=source-sans-pro--numbers`

I must be missing something out, sorry but I am fairly new to docker. Can you please help me out here as well? Thanks a lot!

**Comment by kushaggarwal** [comment]
You need to run the storybook side by side for the test to run on it locally by 
 `npm run storybook`

**Comment by Dishebh** [comment]
> You need to run the storybook side by side for the test to run on it locally by
> `npm run storybook`

Thank you so much! I was actually trying to run the storybook and the tests in the same container simultaneously, but then after some surfing I got to know it was not feasible. 
Thanks a lot, again :)

**Pull request merged**